File: dotfiles/local_bin/+archive+/yesno.py
#!/usr/bin/env python

import sys
import subprocess
import shlex

if __name__ == '__main__':
    title = sys.argv[1] if len(sys.argv) > 1 else 'really???'

    cmd = f"rofi -theme-str 'window {{width: 200px; height: 150px;}}' -dmenu -p '{title}' -sep '\\0' -eh 2 -markup-rows -format i"
    # cmd = shlex.split(cmd)
    # print(cmd)
    with subprocess.Popen(cmd, stdin=subprocess.PIPE, shell=True) as p:
        p.stdin.write(b"<span size='x-large' weight='heavy'>Yes</span>\0")
        p.stdin.write(b"<span size='x-large' weight='heavy'>No</span>\0")
        p.stdin.flush()
        exitcode = p.wait()
        # print('EXIT:', exitcode)

File: dotfiles/local_bin/+archive+/yesno.rb
#!/usr/bin/env ruby

require 'shellwords'
require 'open3'


title = if ARGV.length > 0 then ARGV[0] else 'really???' end

cmd = "rofi -theme-str 'window {width: 200px; height: 150px;}' -dmenu -p '#{title}' -sep '\\0' -eh 2 -markup-rows -format i"
Open3.popen2(cmd) do |i, o, t|
  i.write("<span size='x-large' weight='heavy'>Yes</span>\0")
  i.write("<span size='x-large' weight='heavy'>No</span>\0")

  # pid = t.pid # pid of the started process.
  exit_status = t.value
  p exit_status
  p o.read
end
